### PR TITLE
Expand clean slate

### PR DESCRIPTION
**Pull request by bentheiii**
WIP

### CODE REVIEW

File: yellowbox/extras/rabbit_mq.py [hunk 1]
```diff
@@ -56,3 +60,44 @@ def enable_management(self):
...
+        Notes:
+            This feature is experimental.
+        """
+        try:
+            management_url = self.management_url()
+        except RuntimeError as e:
+            raise RuntimeError('management must be enabled for the service for clean_slate') from e
+
+        queues_url = management_url + 'api/queues'
+        response = requests.get(queues_url, auth=(self.user, self.password))
```
Comment (bharel) [inline yellowbox/extras/rabbit_mq.py:76]: def _get_queues(self)?

File: yellowbox/extras/rabbit_mq.py [hunk 2]
```diff
@@ -56,3 +60,44 @@ def enable_management(self):
...
+        replies = requests.get(queues_url, auth=(self.user, self.password))
+        replies.raise_for_status()
+        extant_queues = replies.json()
+        delete_params = {}
+        if not force_queue_deletion:
+            delete_params['if-unused'] = 'true'
+        for queue in extant_queues:
+            name = quote(queue['name'], safe='')
+            vhost = quote(queue['vhost'], safe='')
+            requests.delete(
```
Comment (bharel) [inline yellowbox/extras/rabbit_mq.py:91]: def _delete_queue(self, name, force=False)?

**Review by bharel — APPROVED** [review]
LGTM

File: yellowbox/extras/rabbit_mq.py (hunk 1 above)
Comment (bentheiii) [reply yellowbox/extras/rabbit_mq.py:76]: seems like a lot of overhead for something that will only be called once

Comment (bharel) [reply yellowbox/extras/rabbit_mq.py:76]: 🤷 might be useful later on. The only thing I'm not sure about is again, whether we should have utility functions on the service or not. Still debating that one.

Comment (bentheiii) [reply yellowbox/extras/rabbit_mq.py:76]: it's experimantal

Comment (bharel) [reply yellowbox/extras/rabbit_mq.py:76]: Not talking about "clean slate". Talking about generally adding application functions to the Service object.

**Pull request merged**